cinema/views.py

```diff
@@ -45,9 +73,31 @@ def get_serializer_class(self):
 
 
 class MovieSessionViewSet(viewsets.ModelViewSet):
-    queryset = MovieSession.objects.all()
+    queryset = (
+        MovieSession.objects.all()
+        .select_related("movie", "cinema_hall")
+        .annotate(
+            tickets_available=F("cinema_hall__rows")
+            * F("cinema_hall__seats_in_row")
+            - Count("tickets")
```

Comment:
The `tickets_available` annotation calculates available tickets by subtracting the count of tickets from the total capacity. Ensure that this logic correctly handles all edge cases, such as when the cinema hall's capacity changes or when tickets are reserved but not yet confirmed.